src/analyzer/analyzersilence.cpp

```diff
@@ -117,29 +117,49 @@ void AnalyzerSilence::storeResults(TrackPointer tio) {
     double introStart = mixxx::kAnalysisChannels * m_iSignalStart;
     double outroEnd = mixxx::kAnalysisChannels * m_iSignalEnd;
 
-    if (shouldUpdateMainCue(tio->getCuePoint())) {
-        tio->setCuePoint(CuePosition(introStart, Cue::AUTOMATIC));
+    if (shouldUpdateMainCue(pTrack->getCuePoint())) {
+        pTrack->setCuePoint(CuePosition(introStart, Cue::Source::Automatic));
     }
 
-    CuePointer pIntroCue = tio->findCueByType(Cue::INTRO);
+    CuePointer pIntroCue = pTrack->findCueByType(Cue::Type::Intro);
     if (!pIntroCue) {
-        pIntroCue = tio->createAndAddCue();
-        pIntroCue->setType(Cue::INTRO);
-        pIntroCue->setSource(Cue::AUTOMATIC);
+        pIntroCue = pTrack->createAndAddCue();
+        pIntroCue->setType(Cue::Type::Intro);
+        pIntroCue->setSource(Cue::Source::Automatic);
     }
-    if (pIntroCue->getSource() != Cue::MANUAL) {
+    if (pIntroCue->getSource() != Cue::Source::Manual) {
         pIntroCue->setPosition(introStart);
         pIntroCue->setLength(0.0);
     }
 
-    CuePointer pOutroCue = tio->findCueByType(Cue::OUTRO);
+    CuePointer pOutroCue = pTrack->findCueByType(Cue::Type::Outro);
     if (!pOutroCue) {
-        pOutroCue = tio->createAndAddCue();
-        pOutroCue->setType(Cue::OUTRO);
-        pOutroCue->setSource(Cue::AUTOMATIC);
+        pOutroCue = pTrack->createAndAddCue();
+        pOutroCue->setType(Cue::Type::Outro);
+        pOutroCue->setSource(Cue::Source::Automatic);
     }
-    if (pOutroCue->getSource() != Cue::MANUAL) {
+    if (pOutroCue->getSource() != Cue::Source::Manual) {
         pOutroCue->setPosition(-1.0);
         pOutroCue->setLength(outroEnd);
     }
+
+    CuePointer pFirstSound = pTrack->findCueByType(Cue::Type::FirstSound);
+    if (pFirstSound == nullptr) {
+        pFirstSound = pTrack->createAndAddCue();
+        pFirstSound->setType(Cue::Type::FirstSound);
+        pFirstSound->setSource(Cue::Source::Automatic);
+        pFirstSound->setPosition(introStart);
+    } else if (pFirstSound->getSource() != Cue::Source::Manual) {
```

Comment:
The else case should actually never happen if FirstSound is hidden. If the source is set to another value than `Cue::Source::Automatic` this is a programming error or data corruption.

```
DEBUG_ASSERT(!pFirstSound || pFirstSound->getSource() == Cue::Source::Automatic);
(!pFirstSound) {
   pFirstSound = pTrack->createAndAddCue();
   pFirstSound->setType(Cue::Type::FirstSound);
}
pFirstSound->setSource(Cue::Source::Automatic);
pFirstSound->setPosition(introStart);
```
no else, i.e. an inconsistent cue point will be fixed implicitly.